### ISSUE TITLE
[New rule] Prefer single widget per file

### ISSUE THREAD

**Issue by incendial** [dart-code-checker/dart-code-metrics#390]
**Please describe what the rule should do:**
The rule should recommend to have a single widget per file.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

Bad:

**some_widget.dart**
```dart
class someWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    ...
  }
}

// LINT
class someOtherWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    ...
  }
}

// LINT
class _someOtherWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    ...
  }
}

// LINT
class someStatefulWidget extends StatefulWidget {
  @override
  _someStatefulWidgetState createState() => _someStatefulWidgetState();
}

class _someStatefulWidgetState extends State<InspirationCard> {
  @override
  Widget build(BuildContext context) {
    ...
  }
}

```

Good:

**some_widget.dart**
```dart
class someWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    ...
  }
}
```

**some_other_widget.dart**
```dart
class someOtherWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    ...
  }
}
```

**Comment by incendial** [comment]
Available in 4.2.0 release.
